.changeset/mighty-horses-fetch.md

```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/cli-kit': patch
+---
+
+Add custom controllable request behaviour for theme related commands
```

Comment:
mega nit: Would love a mention of the problem we are fixing here. This almost implies that users can now control request behaviour